test/python/algorithms/optimizers/test_gradient_descent.py

```diff
@@ -0,0 +1,116 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for the Gradient Descent optimizer."""
+
+from test.python.algorithms import QiskitAlgorithmsTestCase
+
+import numpy as np
+
+from qiskit.algorithms.optimizers import GradientDescent
+from qiskit.circuit.library import PauliTwoDesign
+from qiskit.opflow import I, Z, StateFn
+from qiskit.test.decorators import slow_test
+
+
+class TestGradientDescent(QiskitAlgorithmsTestCase):
+    """Tests for the SPSA optimizer."""
```

Comment:
SPSA is here and in line 34 below. This doc string should be corrected to refer to gradient descent.

Reply:
good catch, thanks!